watertap/unit_models/translators/translator_asm2d_adm1.py

```diff
@@ -395,18 +402,18 @@ def SN_org(blk, t):
 
         @self.Expression(self.flowsheet().time, doc="Monosaccharides mapping")
         def Ssu_mapping(blk, t):
-            return Expr_if(
-                blk.SN_org[t] >= blk.properties_in[t].conc_mass_comp["S_F"],
-                eps * pyunits.kg / pyunits.m**3,
+            return smooth_max(
+                0 * pyunits.kg / pyunits.m**3,
                 blk.properties_in[t].conc_mass_comp["S_F"] - blk.SN_org[t],
+                blk.eps,
             )
```

Comment:
I am a little confused here, I think the original equation, if A>=B, then we have `eps`, but in the the smooth_max function, you directly change this `eps` to 0 if I understand correctly, and the `eps` in the smoothing parameter you set at `1e-10`. My first question is: does change previous `eps` to `0` have no impacts to all of the examples? And why do you set `1e-10` for the smoothing parameter

Reply:
Setting `eps` to 0 means it is no longer smooth, and can thus trigger numerical issues.

Reply:
So there are two instances of `eps` in this file. `eps` is 0 and `blk.eps` is 1e-10. `blk.eps` is the one used in the smooth_max and smooth_min functions. I think I should just directly replace instances of `eps` with zero to make this less confusing

Reply:
@luohezhiming I changed the epsilon used for smooth_max and smooth_min to `eps_smooth` to make things a bit more transparent. Also, the 1e-10 is the smoothing factor used in the ASM1/ADM1 translator, so I just carried that value over. It can be changed if necessary.

Reply:
@MarcusHolly You should not be using `eps=0` in the `smoothmax` and `smoothmin` functions - that defeats the entire purpose if those functions. If you really want a non-smooth max or min then you should refactor these to use `Expr_if`.

Reply:
eps=1e-10 in the smooth_max. Are you saying it shouldn't be that close to 0 @andrewlee94?

Reply:
Probably not - the value you use for `eps` represents how smooth the operation is. Smaller values mean less smoothness and thus more chance for numerical issues (but less numerical error). Ultimately, `eps` needs to be tuned for the application, and should be a few orders of magnitude less than the inputs to the `max` operation.

Reply:
Changed the `eps_smooth` value to 1e-4 since seems to give the best solver performance in the `BSM2_P_extension.py` flowsheet.

Reply:
So the `eps` now only represents the smoothing parameter, we can try to vary the `smooth_eps` if all tests get passed.